herbert_core/classes/MPIFramework/@JobExecutor/JobExecutor.m

```diff
@@ -134,9 +178,55 @@
             %
             [obj,mess]=init_je_(obj,fbMPI,intercom_class,InitMessage,synchronize);
         end
+        %------------------------------------------------------------------
+        function id = get.labIndex(obj)
+            % get number (job id) of current running job
+            if isempty(obj.mess_framework_)
+                % class has not been initiated properly
+                id  = -1;
```

Comment:
if someone is accessing an incorrectly initialised class should this raise an error, or is `-1` recognized elsewhere as a "not initialised" flag?

Reply:
Its abstract class, and this issue is used somewhere in testing (probably)  or not. But anyway, nobody would instantiate it as such. 
-1 is generally either not initialized or `any`  within MPI framework code scope. Non-initialized state is necessary.
"not initialized" is also some flag which is not recognized and needs to be recognized somewhere. 
its not C where you typedef NOT_INITIALIZED and it is there.....
Do not know how much work it will take to resolve to introduce such flag. I tried with MESS_CODES for MPI messages and its not that convenient. 
Unsure, so better leave it as it is.....

Reply:
or write a ticket to have it updated, Ít can be relatively big work.